kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/utils/WriteUtils.scala

```diff
@@ -50,11 +50,19 @@ import org.apache.commons.io.FileUtils
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
 
-/** Default write utilities that use the V1 transaction builders. */
+/** Default write utilities that use the V1 transaction builders and legacy Table Snapshot APIs */
 trait WriteUtils extends AbstractWriteUtils with TransactionBuilderV1Support
+    with TestUtilsWithLegacyKernelAPIs
 
-/** Write utilities that use the V2 transaction builders to create transactions */
+/**
+ * DO NOT MODIFY this trait -- this is just syntactic sugar to clearly indicate we are extending the
+ * "default" WriteUtils which happens to use the legacy Kernel APIs
+ */
+trait WriteUtilsWithV1Builders extends WriteUtils
+
+/** Write utilities that use the V2 transaction builders to create transactions and TableManager snapshot APIs */
 trait WriteUtilsWithV2Builders extends AbstractWriteUtils with TransactionBuilderV2Support
```

Comment:
Note: this change does mean that all other existing write suites that extend `WriteUtilsWithV2Builders` will now run with `TestUtilsWithTableManagerAPIs`. However these suites have not been refactored to use the `getTableManagerAdapter` method so this does not mean they've been fully refactored to use the new read APIs (probably some mix of it, helpers within test utilities may use the new APIs, but the suites may also manually load snapshots)

I think this is a fine state for now!